src/Database/PostgreSQL/PQTypes/DerivingVia.hs

```diff
@@ -0,0 +1,113 @@
+{-# LANGUAGE AllowAmbiguousTypes, TypeApplications #-}
+module Database.PostgreSQL.PQTypes.DerivingVia (
+  -- | Helpers, to be used with `deriving via`.
+    SQLEnum(..)
+  , SQLEnumEncoding(..)
+  , SQLEnumAsText(..)
+  , SQLEnumAsTextEncoding(..)
+  , isInjective  -- ^ for doctests
+  ) where
+
+import Control.Exception (SomeException(..), throwIO)
+import Data.List.Extra (enumerate, nubSort)
+import Data.Map.Strict (Map)
+import Data.Text (Text)
+import Data.Typeable
+import Database.PostgreSQL.PQTypes
+import qualified Data.Map.Strict as Map
+
+-- | To be used in doctests to prove injectivity of encoding functions.
+--
+-- >>> isInjective (id :: Bool -> Bool)
```

Comment:
I've enabled doctests in the CI so these will actually be checked :tada:

Reply:
Yay! Let's see what I need to change to make them pass...